File: lab7/ex15.jl
#!/usr/local/bin/julia
counter = 0

function f(n::Int64)
    s = 0
    if n == 0
        return 1
    else
        for i in 0:n-1
            s += f(i)   # line 6
            # Counting
            global counter
            counter += 1
        end
        return s
    end
end


n_range = 16
if length(ARGS) > 0
    n_range = parse(Int64, ARGS[1])
end
ns = 0:n_range
for n in ns
    s_value = f(n)
    c_value = counter
    global counter
    counter = 0
    println("n = $n: ctr = $c_value; s = $s_value")
end

File: lab7/ex15.py
#!/usr/bin/python3
from sys import argv

counter = 0

def f(n: int) -> int:
    s = 0
    if n == 0:
        return 1
    else:
        for i in range(n):
            s += f(i)   # line 6
            # Counting
            global counter
            counter += 1
        return s
    

if __name__ == "__main__":
    n_range = 16
    if len(argv) > 1:
        n_range = int(argv[1])
    ns = range(n_range+1)
    for n in ns:
        s_value = f(n)
        c_value = counter
        counter = 0
        print(f"n = {n}: ctr = {c_value}; s = {s_value}")

